statsmodels/imputation/bayes_mi.py

```diff
@@ -66,7 +66,13 @@ def __init__(self, data, mean_prior=None, cov_prior=None, cov_prior_df=1):
             if v not in rowmap:
                 rowmap[v] = []
             rowmap[v].append(i)
-        self.patterns = [np.asarray(x) for x in rowmap.values()]
+        patterns = [np.asarray(v) for v in rowmap.values()]
+
+        # Give the missing data patterns a deterministic order
```

Comment:
It might be easier just to use a ordereddict for rowmap
or sort by rowmap values (sorting list of list should work and it will be just a one time setup cost)